packages/mdc-switch/_mixins.scss

```diff
@@ -23,56 +23,287 @@
+      @include mdc-theme-prop(background-color, $color);
+      @include mdc-theme-prop(border-color, $color);
+    }
   }
 }
 
-@mixin mdc-switch-toggled-on-ripple-color($color) {
+@mixin mdc-switch-toggled-on-ripple-color($color, $query: mdc-feature-all()) {
   &.mdc-switch--checked .mdc-switch__thumb-underlay {
-    @include mdc-states($color);
+    @include mdc-states($color, false, $query);
   }
 }
 
-@mixin mdc-switch-toggled-off-track-color($color) {
+@mixin mdc-switch-toggled-off-track-color($color, $query: mdc-feature-all()) {
+  $feat-color: mdc-feature-create-target($query, color);
+
   &:not(.mdc-switch--checked) .mdc-switch__track {
-    @include mdc-theme-prop(background-color, $color);
-    @include mdc-theme-prop(border-color, $color);
+    @include mdc-feature-targets($feat-color) {
+      @include mdc-theme-prop(background-color, $color);
+      @include mdc-theme-prop(border-color, $color);
+    }
   }
 }
 
-@mixin mdc-switch-toggled-off-thumb-color($color) {
+@mixin mdc-switch-toggled-off-thumb-color($color, $query: mdc-feature-all()) {
+  $feat-color: mdc-feature-create-target($query, color);
+
   &:not(.mdc-switch--checked) .mdc-switch__thumb {
-    @include mdc-theme-prop(background-color, $color);
-    @include mdc-theme-prop(border-color, $color);
+    @include mdc-feature-targets($feat-color) {
+      @include mdc-theme-prop(background-color, $color);
+      @include mdc-theme-prop(border-color, $color);
+    }
   }
 }
 
-@mixin mdc-switch-toggled-off-ripple-color($color) {
+@mixin mdc-switch-toggled-off-ripple-color($color, $query: mdc-feature-all()) {
   &:not(.mdc-switch--checked) .mdc-switch__thumb-underlay {
-    @include mdc-states($color);
+    @include mdc-states($color, false, $query);
+  }
+}
+
+//
+// Private
+//
+
+// Structure
+@mixin mdc-switch-base_ {
+  display: inline-block;
+  position: relative;
+  outline: none;
+  user-select: none;
+}
+
+@mixin mdc-switch__track_($query: mdc-feature-all()) {
+  $feat-animation: mdc-feature-create-target($query, animation);
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-feature-targets($feat-structure) {
+    box-sizing: border-box;
+    width: $mdc-switch-track-width;
+    height: $mdc-switch-track-height;
+    border: 1px solid;
+    border-radius: $mdc-switch-track-height / 2;
+    opacity: .38;
+  }
+
+  @include mdc-feature-targets($feat-animation) {
+    transition:
+      mdc-switch-transition(opacity),
+      mdc-switch-transition(background-color),
+      mdc-switch-transition(border-color);
+  }
+}
+
+@mixin mdc-switch__thumb-underlay_($query: mdc-feature-all()) {
+  $feat-animation: mdc-feature-create-target($query, animation);
+  $feat-color: mdc-feature-create-target($query, color);
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-ripple-surface($query);
+  @include mdc-states($mdc-switch-baseline-theme-color, false, $query);
+  @include mdc-ripple-radius-unbounded(100%, $query);
```

Comment:
Move this above `mdc-states`

Reply:
Whoops. Ignore this comment; it was hanging around as a pending comment but I already resolved it on the branch when I linted it.